server/src/main/java/ru/otus/ohmyval/homeworks/chat/server/DatabaseAuthenticationService.java

```diff
@@ -0,0 +1,226 @@
+
+public class DatabaseAuthenticationService implements AuthenticationService {
+    private class User {
+        private int id;
+        private String login;
+        private String password;
+        private String nickname;
+        private Role role;
+
+        public int getId() {
+            return id;
+        }
+
+        public String getLogin() {
+            return login;
+        }
+
+        public String getPassword() {
+            return password;
+        }
+
+        public String getNickname() {
+            return nickname;
+        }
+
+        public Role getRole() {
+            return role;
+        }
+
+        private void setRole(Role role) {
+            this.role = role;
+        }
+
+        public User(int id, String login, String password, String nickname) {
+            this.id = id;
+            this.login = login;
+            this.password = password;
+            this.nickname = nickname;
+        }
+    }
+
+    public class Role {
+        private int id;
+        private String title;
+
+        public Role(int id, String title) {
+            this.id = id;
+            this.title = title;
+        }
+
+        public int getId() {
+            return id;
+        }
+
+        public String getTitle() {
+            return title;
+        }
+    }
+
+
+    private static final String DATABASE_URL = "jdbc:postgresql://localhost:5432/chat";
+    private static final String USERS_ADD_QUERY = "INSERT INTO users (login, password, nickname) values ('login' + ?, 'pass' + ?, 'nick' + ?)";
+    private static final String USER_REGISTER_QUERY = "INSERT INTO users (login, password, nickname) values (?, ?, ?)";
+    private static final String USER_ROLE_ADMIN_QUERY = "INSERT INTO user_role (user_id, role_id) values (?, '1')";
+    private static final String USER_ROLE_USER_QUERY = "INSERT INTO user_role (user_id, role_id) values (?, '2')";
+    private static final String USER_ROLE_QUERY = "select r.id as id, r.title as title from user_role ur left join roles r ON r.id=ur.role_id where ur.user_id = ?";
+
+    private List<User> users;
+
+    public DatabaseAuthenticationService() {
+        this.users = new ArrayList<>();
+        databaseOperation();
+    }
+
+    public void databaseOperation() {
+        try (Connection connection = DriverManager.getConnection(DATABASE_URL, "postgres", "")) {
+            fillUsers(connection);
+            setRoleAdmin(connection);
+            setRoleUser(connection);
+            extractUserRoles(connection);
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+    }
+    private void fillUsers(Connection connection) throws SQLException {
+        try (PreparedStatement ps = connection.prepareStatement(USERS_ADD_QUERY)) {
+            for (int i = 1; i <= 10; i++) {
+                ps.setInt(1, i);
+                ps.setInt(2, i);
+                ps.setInt(3, i);
+                try (ResultSet usersResultSet = ps.executeQuery()) {
```

Comment:
it's incorrect use for executeQuery with INSERT request because INSERT doesn't return any result.

you should use executeQuery only with SELECT requests.

Reply:
исправила